Automattic/woocommerce-payments
# fix: error message on 402 status code

Fixes https://github.com/Automattic/woocommerce-payments/issues/8079 

#### Changes proposed in this Pull Request









Stripe will sometimes return a 402 error code when a payment fails.
402 is problematic on some hosting (well, the only one we're aware of is Atomic & JN). So let's just return `400` when a `402` is returned by Stripe.

There are also other error codes that have custom responses, but let's start with `402`: p1707297315719429-slack-C7YPW6K40

The argument for returning the `400` is: if Stripe returns a `500` error, would our plugin need to return a `500` error?
Our plugin worked fine, and the communication with Stripe worked fine as well.

#### Testing instructions





You'll need an atomic site or a JN site to test the symptoms:
- Go to "My account > Add payment method", use the card `4000000000000002`
- The screenshot below should no longer occur
<img width="830" alt="Screenshot 2024-03-22 at 1 36 12 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/60ae9895-48bd-42af-ad4e-4c302f8133c6">
- Instead, we should see:
<img width="710" alt="Screenshot 2024-03-22 at 1 36 54 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/b05278bf-c4f0-43b9-a1d3-0fa44cb2b539">

-------------------


**Post merge**
